### PR TITLE
machines: fix missing checks for 0 storage pools in disk add dialog and add option to change disk cache policy 

### PR DESCRIPTION
**Pull request by KKoukiou**

### CODE REVIEW

**Comment by KKoukiou** [comment]
Whoever decides to review this one, let me know if you prefer that I will split it into 2 PRs, since the 2 commits can be separated.

File: pkg/machines/components/diskAdd.jsx [hunk 1]
```diff
@@ -208,6 +209,49 @@ const PoolRow = ({ idPrefix, onValueChanged, storagePoolName, vmStoragePools })
...
 };
 
+class PerformanceOptions extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = { expanded: false };
+    }
+
+    render() {
+        const cacheModes = ['default', 'none', 'writethrough', 'writeback', 'directsync', 'unsafe'];
```
Comment (Gundersanne) [inline pkg/machines/components/diskAdd.jsx]: Any way not to hard code those/retrieve them from some spec? Otherwise it's fine.

Comment (Gundersanne) [reply pkg/machines/components/diskAdd.jsx]: Ah ya, that's alright then.

**Comment by KKoukiou** [comment]
disk-cache

**Pull request merged**